internal/filecache/file_cache.go

```diff
@@ -69,12 +69,28 @@ func (fc *fileCache) Add(key Key, content io.Reader) (err error) {
 	fc.mux.Lock()
 	defer fc.mux.Unlock()
 
-	file, err := os.Create(fc.path(key))
+	// Use rename for an atomic write (on Unix, at least; Windows needs special direct syscall code)
```

Comment:
Since https://go.dev/cl/6140 I'd say `os.Rename` is "atomic enough" on Windows (using `MoveFileEx`), so I don't think we really need this comment.